### DISCUSSION TITLE
Problem loading the firmware

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#534] [Q&A] [answered]
Hi.

The reason I am writing to you is to ask if there is a way to check if the crazyflie actually has any of the controls loaded (PID, INDI or Mellinger)?, Since I have made some changes in the codes and when I carry out tests, Crazyflie continues workIing as if these modifications did not exist.

I would appreciate your help.

Regards.

**Comment by krichardsson** [comment, marked as answer]
Hi!

All the controllers are part of the firmware and are not loaded dynamically, that is they are already "loaded". To change controller you use a parameter ([stabilizer.controller](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/params/#stabilizercontroller)) to tell the system which controller to use.

**Reply by Wallat1** [reply]
Hi.

Thanks for answering, thanks to the cfclient interface, I was able to check if crazyflie really has the control that I'm loading.

Regards.
